### PR TITLE
Task 5

### PR DESCRIPTION
**Pull request by Max-Korsakov**
See README for tasks description.
Scrin for 5.2 task (importFileParser log)
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,23 +8,17 @@ CloudFront Adress
 
 ## Tasks
 
-- 1 - Task 4.1 is implemented - done (see add-data-scripts.sql)
-- 2 - TASK 4.2 is implemented lambda links are provided and returns data - done
-- 3 - TASK 4.3 is implemented lambda links are provided and products is stored in DB (call TASK 4.2 to see the product)
-- 4 - Your own Frontend application is integrated with product service (/products API) and products from product-service are represented on Frontend. Link to a working Front-End application is provided for cross-check reviewer.
+Reviewers should verify the lambda functions by invoking them through provided URLs.
```
Comment (Panibrat) [inline README.md:11]: Please provide a link to your FE application at Pull request description

File: import-service/handler.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import "source-map-support/register";
+
+import importProductsFile from "./handlers/importProductsFile";
```
Comment (Panibrat) [inline import-service/handler.ts:3]: ```export  importProductsFile  from './handlers/importProductsFile';``` Just simplify :)

File: import-service/handlers/importFileParser.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import * as csv from "csv-parser";
+
+import { s3 } from "../constants";
+
+const importFileParser: APIGatewayProxyHandler = async (
+  event: APIGatewayProxyEvent,
+  _context: Context
+): Promise<APIGatewayProxyResult> => {
+  try {
+    await Promise.all(
```
Comment (Panibrat) [inline import-service/handlers/importFileParser.ts:17]: 👍

File: import-service/handlers/importProductsFile.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    ContentType: "text/csv",
+  };
+  try {
+    let signedUrlPut = await s3.getSignedUrl("putObject", params);
+    return {
+      statusCode: 200,
+      headers: {
+        "Access-Control-Allow-Origin": "*",
+      },
+      body: JSON.stringify(signedUrlPut),
```
Comment (Panibrat) [inline import-service/handlers/importProductsFile.ts:31]: You are receiving a string. It is not necessary to stringify it.

File: import-service/serverless.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  provider: {
+    name: "aws",
+    runtime: "nodejs12.x",
+    stage:"dev",
+    region: "eu-west-1",
+    iamRoleStatements: [
+      {
+        Effect: "Allow",
+        Action: "s3:*",
+        Resource: "arn:aws:s3:::node-aws-import",
```
Comment (Panibrat) [inline import-service/serverless.ts:28]: ```Resource: [`arn:aws:s3:::${BUCKET}`]``` Could you save the name of the bucket in a CONSTANT?

File: README.md (hunk 1 above)
Comment (Max-Korsakov) [reply README.md:11]: As I mentioned above, see reedme file, please
https://github.com/Max-Korsakov/nodejs-aws-be/blob/task-5/README.md

Comment (Max-Korsakov) [reply README.md:11]: image

**Pull request merged**